### ISSUE TITLE
Grave Items lost when dying during grave retrieval

### ISSUE THREAD

**Issue by NillerMedDild** [Quarris/EnigmaticGraves#21]
Relayed from: https://github.com/NillerMedDild/Enigmatica6/issues/2986

>  died and picked up my gravestone at the same time. Then I went back to the gravestone I just created and picked it up, the items in the inventory and armor slots which should be stored in the gravestone disappeared. Only the baubles (ex: backpack, necklace etc.) remained.

### Versions
- Enigmatica 6 v0.5.9
- [EnigmaticGraves-1.5.1.jar](https://www.curseforge.com/minecraft/mc-mods/enigmatic-graves/files/3402632)
- Forge-1.16.5-36.2.1

**Comment by Quarris** [comment]
Anything in the server logs that could be relevant?

**Comment by NillerMedDild** [comment]
Asked for logs, will relay asap 👍

**Comment by Quarris** [comment]
Also they didn't mention if they tried retrieving grave via commands. Would be handy to know if that worked

**Comment by NillerMedDild** [comment]
> I was single player, but I still upload the logs.
> https://drive.google.com/file/d/1yDjaDqKCOgxwAKaRWJgSetbFKqzuzhuC/view?usp=sharing

I asked about command restoration, awaiting answer :)

**Comment by Quarris** [comment]
None of the crash logs indicate that the player has died. They all must be old I think :/

**Comment by NillerMedDild** [comment]
Answer from OP:

> Sorry, I don't know which command I should use to restore the grave. On the other hand, I updated the version recently, so I think I got the versions mixed up. It should be Enigmatica6-0.5.8.
It is the log. https://drive.google.com/file/d/1JsDjmXrEu1jls-ypYeZlV2bsi5y_-KCT/view?usp=sharing
The incident happened after 2021-8-10. (Sorry, I don't remember the exact date)

**Comment by NillerMedDild** [comment]
I'm gonna close the issue on E6, I don't have time to look through that many logs for an extreme edge case bug :/

**Comment by Quarris** [comment]
I'm gonna have a quick look but if I don't find anything useful I won't be able to do it sadly. Its a whole different story if the command isn't working as well

**Comment by Quarris** [comment]
It is also possible I'm gonna need to add more logging for retrieving up graves. Because if that happened when they retrieved and died at the same time I still wouldn't be able to distinguish between a normal death and that bug :/

**Comment by NillerMedDild** [comment]
He didn't try to use the command, so I don't know :<

**Comment by DeeDoubleYous** [comment]
I can replicate this in Enigmatica 6 - 0.5.20 which uses Enigmatic Graves 1.6.5. What's happening is when you return to the grave you click on it to get the contents and in the same second, you die. You get the items in your inventory as the death screen shows and a new grave is spawned in place with just the grave locator of the old grave present.  I encountered this when an Atum mob for some reason was spawned into the overworld and camped the graves and killed me as I clicked on the grave when I used /back to get the grave.
https://youtu.be/9YwNQbiCJ1A here's a video showing the issue.
